### PR TITLE
COMMERCE-9999

### PR DESCRIPTION
**Pull request by brianikim**
https://issues.liferay.com/browse/COMMERCE-9999

cc/ @andrea-ale-sbarra , @alessiorendina

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart/view.jsp [hunk 1]
```diff
@@ -101,10 +103,17 @@ Map<Long, List<CommerceOrderValidatorResult>> commerceOrderValidatorResultsMap =
 					>
 						<span class="sticker sticker-xl">
 							<span class="sticker-overlay">
-								<liferay-adaptive-media:img
-									class="sticker-img"
-									fileVersion="<%= commerceCartContentDisplayContext.getCPInstanceImageFileVersion(commerceOrderItem) %>"
-								/>
+								<c:choose>
+									<c:when test="<%= !cpInstanceCDNURL.equals(StringPool.BLANK) %>">
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart/view.jsp:107]: Use `Validator.isNotNull(cpInstanceCDNURL)`

File: modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp [hunk 2]
```diff
@@ -105,17 +105,26 @@ request.setAttribute("view.jsp-portletURL", portletURL);
...
 					cssClass="col-1 thumbnail-section"
 				>
 					<span class="sticker sticker-xl">
 						<span class="sticker-overlay">
-							<liferay-adaptive-media:img
-								class="sticker-img"
-								fileVersion="<%= commerceCartContentMiniDisplayContext.getCPInstanceImageFileVersion(commerceOrderItem) %>"
-							/>
+							<c:choose>
+								<c:when test="<%= !cpInstanceCDNURL.equals(StringPool.BLANK) %>">
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp:118]: Same here

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp [hunk 3]
```diff
@@ -95,10 +97,17 @@ Map<Long, List<CommerceOrderValidatorResult>> commerceOrderValidatorResultsMap =
 						>
 							<span class="sticker sticker-xl">
 								<span class="sticker-overlay">
-									<liferay-adaptive-media:img
-										class="sticker-img"
-										fileVersion="<%= orderSummaryCheckoutStepDisplayContext.getCPInstanceImageFileVersion(commerceOrderItem) %>"
-									/>
+									<c:choose>
+										<c:when test="<%= !cpInstanceCDNURL.equals(StringPool.BLANK) %>">
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp:101]: Same here

File: modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_publisher/render/list/entry/view.jsp [hunk 4]
```diff
@@ -32,10 +34,17 @@ String productDetailURL = cpContentHelper.getFriendlyURL(cpCatalogEntry, themeDi
 	<div class="card d-flex flex-column product-card">
 		<div class="card-item-first position-relative">
 			<a href="<%= productDetailURL %>">
-				<liferay-adaptive-media:img
-					class="img-fluid product-card-picture"
-					fileVersion="<%= cpContentHelper.getCPDefinitionImageFileVersion(cpCatalogEntry.getCPDefinitionId(), request) %>"
-				/>
+				<c:choose>
+					<c:when test="<%= !cpDefinitionCDNURL.equals(StringPool.BLANK) %>">
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_publisher/render/list/entry/view.jsp:38]: Same here

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@brianikim Good job, just a few comments, please check them. Thank you!

**Comment by brianikim** [comment]
@alessiorendina , changed the commits as requested with the changes.

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp [hunk 5]
```diff
@@ -105,17 +105,26 @@ request.setAttribute("view.jsp-portletURL", portletURL);
...
 					cssClass="col-1 thumbnail-section"
 				>
 					<span class="sticker sticker-xl">
 						<span class="sticker-overlay">
-							<liferay-adaptive-media:img
-								class="sticker-img"
-								fileVersion="<%= commerceCartContentMiniDisplayContext.getCPInstanceImageFileVersion(commerceOrderItem) %>"
-							/>
+							<c:choose>
+								<c:when test="<%= Validator.isNotNull(cpInstanceCDNURL) %>">
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp:118]: Ehi @brianikim and @alessiorendina why this changes weren't made also in the react minicart component?

Comment (brianikim) [reply modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp:118]: @andrea-ale-sbarra , could you direct me to the correct file you are referencing?

Comment (brianikim) [reply modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp:118]: @andrea-ale-sbarra , https://github.com/liferay-commerce/liferay-portal/blob/4d087767da18b2fbdeea48b705a9dc12b623e594/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItem.js#L55 , the change you are requesting is not necessary because that code utilizes the back-end code of https://github.com/liferay-commerce/liferay-portal/pull/3044/commits/329a20067e39fdf6251df31b63f3d5958731b446 which fixes the issue. See use-case: https://github.com/liferay-commerce/liferay-portal/blob/4d087767da18b2fbdeea48b705a9dc12b623e594/modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/dto/v1_0/CartItemDTOConverter.java#L85

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-cart-content-web/src/main/resources/META-INF/resources/cart_mini/view.jsp:118]: Alright thank you @brianikim

**Review by andrea-ale-sbarra — APPROVED** [review]
